wgsl/index.bs

```diff
@@ -1368,86 +1372,385 @@ Therefore, a data value must not be placed in the padding at the end of a struct
 
-## Pointer Types TODO ## {#pointer-types}
+## Memory View Types ## {#memory-view-types}
+
+In addition to calculating with [=plain types|plain=] values, a [SHORTNAME] program will
+also often read values from memory or write values to memory.
+Operations that read or write to memory are called <dfn noexport>memory accesses</dfn>.
+Each memory access is performed via a [=memory view=].
+
+A <dfn noexport>memory view</dfn> is a set of [=memory locations=] in a particular [=storage class=],
+together with an interpretation of the contents of those locations as a [SHORTNAME] [=type=].
+
+
+[SHORTNAME] has two kinds of types for representing memory views:
+[=reference types=] and [=pointer types=].
+
 <table class='data'>
   <thead>
-    <tr><th>Type<th>Description
+    <tr><th>Constraint<th>Type<th>Description
   </thead>
-  <tr><td>ptr<*SC*,*T*><td>Pointer (or reference) to storage in [=storage class=] *SC*
-                            which can hold a value of the [=storable=] *T*.
-                            Here, *T* is the known as the *pointee* type.
+  <tr algorithm="memory reference type">
+    <td>|SC| is a [=storage class=],<br>|T| is a [=storable=] type
+    <td>ref&lt;|SC|,|T|&gt;
+    <td>The <dfn noexport>reference type</dfn>
+        identified with the set of [=memory views=] for memory locations in |SC| holding values of type |T|.<br>
+        In this context |T| is known as the <dfn noexport>store type</dfn>.<br>
+        Reference types are not written [SHORTNAME] progam source; instead they are used to analyze a [SHORTNAME] program.
+  <tr algorithm="pointer type">
+    <td>|SC| is a [=storage class=],<br>|T| is a [=storable=] type
+    <td>ptr&lt;|SC|,|T|&gt;
+    <td>The <dfn noexport>pointer type</dfn> 
+        identified with the set of [=memory views=] for memory locations in |SC| holding values of type |T|.<br>
+        In this context |T| is known as the <dfn noexport>pointee type</dfn>.<br>
+        Pointer types appear in [SHORTNAME] progam source.
 </table>
 
-Note: We've described a SPIR-V logical pointer type.
+<div class='example wgsl' heading='Pointer type (program fragment)'>
+  <xmp highlight='rust'>
+    // The type of a pointer value that references storage for keeping
+    // an 'i32' value, using memory locations in the 'uniform' storage class.
+    // Here 'i32' is the pointee type.
+    ptr<uniform, i32>              // Note: This is not a valid expression on its own.
+
+    // The type of a pointer value that references storage for keeping an array
+    // of 50 elements of type 'f32', using memory locations in the 'private'
+    // storage class.
+    // Here the pointee type is 'array<f32,50>'.
+    ptr<private, array<f32, 50>>   // Note: This is not a valid expression on its own.
+  </xmp>
+</div>
 
-Note: Pointers are not storable.
+Reference types and pointer types are both sets of memory views:
+a particular memory view is associated with a unique reference value and also a unique pointer value:
```

Comment:
We're not talking about a comparison being done in WGSL itself, but by the (mathematical) definition of memory views.

A particular "memory view" is a mathematical object that corresponds to two WGSL values: a reference value and a pointer value.   Then use duck-uniqueness:  if two memory views have the same memory locations and same WGSL interpretation type, then they are the same memory views.

Reply:
In fact, the colon at the end of that sentence points calls out the block-quote, which ends with " .... where p and r describe the same memory view".   I think that's clear?

Reply:
sorry, I don't get this still. You are saying that this talking about unique memory views, but what is written here is "unique reference value" and "unique pointer value".
Would it be possible to reword this to avoid ambiguity?